rpmlint/pkg.py

```diff
@@ -759,26 +759,145 @@ def __init__(self, name, is_source=False):
         self.files = {}
         self.ghost_files = {}
 
+        # header is a dictionary to mock rpm metadata
+        self.header = {
+            rpm.RPMTAG_REQUIRENAME: [],
+            rpm.RPMTAG_REQUIREFLAGS: [],
+            rpm.RPMTAG_REQUIREVERSION: [],
+            rpm.RPMTAG_CONFLICTNAME: '',
+            rpm.RPMTAG_CONFLICTFLAGS: '',
+            rpm.RPMTAG_CONFLICTVERSION: '',
+            rpm.RPMTAG_PROVIDENAME: '',
+            rpm.RPMTAG_PROVIDEFLAGS: '',
+            rpm.RPMTAG_PROVIDEVERSION: '',
+            rpm.RPMTAG_OBSOLETENAME: '',
+            rpm.RPMTAG_OBSOLETEFLAGS: '',
+            rpm.RPMTAG_OBSOLETEVERSION: '',
+            rpm.RPMTAG_RECOMMENDNAME: '',
+            rpm.RPMTAG_RECOMMENDFLAGS: '',
+            rpm.RPMTAG_RECOMMENDVERSION: '',
+            rpm.RPMTAG_SUGGESTNAME: '',
+            rpm.RPMTAG_SUGGESTFLAGS: '',
+            rpm.RPMTAG_SUGGESTVERSION: '',
+            rpm.RPMTAG_ENHANCENAME: '',
+            rpm.RPMTAG_ENHANCEFLAGS: '',
+            rpm.RPMTAG_ENHANCEVERSION: '',
+            rpm.RPMTAG_SUPPLEMENTNAME: '',
+            rpm.RPMTAG_SUPPLEMENTFLAGS: '',
+            rpm.RPMTAG_SUPPLEMENTVERSION: '',
+        }
+
+    def _gather_aux(self, header, xs, nametag, flagstag, versiontag,
```

Comment:
This is the same function that we've in the `Pkg` class, so we can move to the `AbstractPkg` class and remove the code from the `Pkg` class, so any child class will inherit the same code.